library/core/src/num/mod.rs

```diff
@@ -661,6 +661,74 @@ impl u8 {
     pub const fn is_ascii_control(&self) -> bool {
         matches!(*self, b'\0'..=b'\x1F' | b'\x7F')
     }
+
+    /// Converts a value to a digit in the given radix.
+    ///
+    /// A 'radix' here is sometimes also called a 'base'. A radix of two
+    /// indicates a binary number, a radix of ten, decimal, and a radix of
+    /// sixteen, hexadecimal, to give some common values. Arbitrary
+    /// radices are supported.
+    ///
+    /// 'Digit' is defined to be only the following ASCII characters:
+    ///
+    /// * `0-9`
+    /// * `a-z`
+    /// * `A-Z`
+    ///
+    /// # Errors
+    ///
+    /// Returns `None` if the value does not refer to a digit in the given radix.
+    ///
+    /// # Panics
+    ///
+    /// Panics if given a radix larger than 36.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// #![feature(to_ascii_digit)]
+    /// assert_eq!(b'1'.to_ascii_digit_radix(10), Some(1));
+    /// assert_eq!(b'f'.to_ascii_digit_radix(16), Some(15));
+    /// ```
+    ///
+    /// Passing a non-digit results in failure:
+    ///
+    /// ```
+    /// #![feature(to_ascii_digit)]
+    /// assert_eq!(b'f'.to_ascii_digit_radix(10), None);
+    /// assert_eq!(b'z'.to_ascii_digit_radix(16), None);
+    /// ```
+    ///
+    /// Passing a large radix, causing a panic:
+    ///
+    /// ```should_panic
+    /// #![feature(to_ascii_digit)]
+    /// // this panics
+    /// b'1'.to_ascii_digit_radix(37);
+    /// ```
+    #[unstable(feature = "to_ascii_digit", issue = "83447")]
+    #[inline]
+    pub fn to_ascii_digit_radix(&self, radix: u32) -> Option<u32> {
```

Comment:
Why is the return value a `u32`?

Reply:
Oh, I was unsure about this as well; it seems weird that the char version also operates on u32 since the max that can be accepted/returned is 36/35, but I figured maybe there was a reason other than than the fact that char==u32 (like '32-bit is a good default choice' or something)

Reply:
Yeah we usually use `u32` as a default for (unsigned) integers that don't get large. E.g. the exponent in `pow` or the return value of `count_zeros`.